### PR TITLE
Fix: Flip order of PR checklist and details

### PR DESCRIPTION
**Pull request by rlmoser99**
Because
If the details are first, they are viewable in Discord's webhook

This PR
* Moves the detail sections to the beginning of the template
* Moves the checklist section to the end of the template
* Adds an issue section to link the issue
* Adds the Because, This PR, and Issue sections to checklist
* Reduce wording that is not essential

Additional Information:
Related to https://github.com/TheOdinProject/curriculum/pull/24779

Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behaviour
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes

Complete the following checkbox ONLY IF it is applicable to your PR. You can complete it later if it is not currently applicable:
-   [ ] I have verified all tests and linters pass against my changes, and/or I have included automated tests where applicable

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (ChargrilledChook) [inline .github/PULL_REQUEST_TEMPLATE.md:26]:
```suggestion
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
```
Minor typo

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (ChargrilledChook) [inline .github/PULL_REQUEST_TEMPLATE.md:36]: For the app repo specifically, I think we should also add `-  [ ] I have squashed my commits`

**Review by ChargrilledChook — COMMENTED** [review]
Few minor comments, otherwise LGTM.

**Comment by KevinMulhern** [comment]
Nothing to add other than thank you for for simplifying this @rlmoser99 🎉  looks great

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rlmoser99) [reply .github/PULL_REQUEST_TEMPLATE.md:36]: I'm not sure that I agree with adding this checkbox. I think it is an expectation of maintainers to do this, but not outside contributors. If we want outside contributors to do this, I think I would like to see an extended set of instructions on how to do this in the contributors guide. 

In addition, I think moving forward, we can use people's "Because" and "This PR" to be their commit message when we rebase & merge in their PR (which I believe has been recently enabled on this repo).

Comment (rlmoser99) [reply .github/PULL_REQUEST_TEMPLATE.md:36]: @KevinMulhern Do you have a preference for adding a checkbox for "I have squashed my commits" (and putting more details in our contributing guide) or leaving it off for the time being and seeing how it goes with us being able to rebase & merge PRs?

Comment (ChargrilledChook) [reply .github/PULL_REQUEST_TEMPLATE.md:36]: I'll defer to Kevin but I think this is an improvement either way, so not a blocker for me.

Definitely worth a bit more disucssion though - following that format is a good convention and it's more valuable the more consistent it is. But maybe it creates an extra barrier to entry for contribution. On balance I would be for forcing it as a requirement for all contributions, but you're right that we would probably want to flesh out the instructions.

[ This page in the wiki ](https://github.com/TheOdinProject/theodinproject/wiki/Squashing-Commits) covers most of it, but we could probably make that page more obvious.

**Review by ChargrilledChook — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rlmoser99) [reply .github/PULL_REQUEST_TEMPLATE.md:36]: Yes, that page does cover it nicely, however if doesn't work that way for me anymore. I don't remember if it is because I've set up oh my zsh or made VS Code my default git editor. (the order is reversed, so I need to have the bottom one as pick) But yes, this does take a bit more discussion, so I'll merge in this PR for now and we'll come back to this in the future.
Screen Shot 2022-10-21 at 9 14 51 PM

Comment (KevinMulhern) [reply .github/PULL_REQUEST_TEMPLATE.md:36]: > Do you have a preference for adding a checkbox for "I have squashed my commits" (and putting more details in our contributing guide) or leaving it off for the time being and seeing how it goes with us being able to rebase & merge PRs?

I think it would be worth adding as a prompt/reminder to squash. But squash and rebase has been working very nicely lately with PR's from outside the maintainer team.

**Pull request merged**
